rec/models/recommender.py

```diff
@@ -332,18 +359,11 @@ def generate_recommendations(self, k=1, indices_prime=None):
         self.log("Row:\n" + str(row))
         self.log("Indices_prime:\n" + str(indices_prime))
         s_filtered = self.predicted_scores[row, indices_prime]
+        # scores are U x I; we can use argsort to sort the item indices
+        # from low to high scores
         permutation = s_filtered.argsort()
         rec = indices_prime[row, permutation]
-        probabilities = np.logspace(0.0, rec.shape[1] / 10.0, num=rec.shape[1], base=2)
-        probabilities = probabilities / probabilities.sum()
-        self.log("Items ordered by preference for each user:\n" + str(rec))
-        picks = self.random_state.choice(
-            permutation.shape[1], p=probabilities, size=(self.num_users, k)
-        )
-        return rec[
-            np.repeat(self.actual_users._user_vector, k).reshape((self.num_users, -1)),
-            picks,
-        ]
+        return rec[:, -k:]
```

Comment:
@elucherini Note that I have intentionally removed the randomness from how the recommender system recommends items for a couple reasons:

1) It makes it easier to test whether the recommender system is actually recommending items that users prefer.
2) It removes the arbitrary log space probability of which item gets shown to the user
3) It guarantees that the recommender system does not show the user multiples of the same item in one iteration. (I'm confused as to why we do that in the first place - the parameter `num_items_per_iter` implies that each user gets shown some # of unique items, but in the current system, it's possible that they are shown the same item repeated `num_items_per_iter` times.)

Context: I was running into problems while trying to test dynamic user preferences b/c I was relying on the recommender system to recommend the most relevant items to each user, and was getting confused when it wasn't doing that. Turns out it's because even when you specify `num_items_per_iter=10` and there are 10 items total, it's not guaranteed that the user will get shown all 10 items, which is a bit confusing.